src/Solvers/matrix_solver_utils.jl

```diff
@@ -154,12 +154,39 @@ function compute_matrix_for_linear_operation(::CPU, template_field, linear_opera
     return A
 end
 
+@kernel function compute_matrix_for_linear_operation_kernel!(Aeᵢⱼₖ, eᵢⱼₖ, linear_operation!, indices, values, args...)
+    i, j, k = @index(Global, NTuple)
```

Comment:
you are launching the kernel with a linear index:
```julia
kernel! = compute_matrix_for_linear_operation_kernel!(device(GPU()), min(256, N), N)
```
where `N` is a scalar, so there is only one index 
```julia
i = @index(Global, Linear)
```
if you want to have 3 indices you have to launch the kernel with a 3-tuple 
```julia
kernel! = compute_matrix_for_linear_operation_kernel!(device(GPU()), min(256, N), (Nx, Ny, Nz))
```

Reply:
Thanks for pointing out! I have made the update now.